code/processes/wdb.q

```diff
@@ -286,12 +345,14 @@ endofdaysortdate:{[dir;pt;tablist;hdbsettings]
     ];
   };
 
-merge:{[dir;pt;tableinfo;mergelimits;hdbsettings;mergemethod]    
+merge:{[dir;pt;tableinfo;mergelimits;hdbsettings;mergemethod;writedownmode]
   setcompression[hdbsettings[`compression]];
   /- get tablename
   tabname:tableinfo[0];
-  /- get list of partition directories for specified table 
-  partdirs:` sv' tabledir,/:k:key tabledir:.Q.par[hsym dir;pt;tabname];
+  /- get list of partition directories for specified table - partbyattr and partbyenum use different folder structure
+  partdirs:$[writedownmode in `partbyenum;
+			 p where 0<count each key each p:` sv' ((-1_` vs p),/:key p:.Q.par[hsym dir;pt;`]),\: tabname;
+			 ` sv' tabledir,/:key tabledir:.Q.par[hsym dir;pt;tabname]];
```

Comment:
indentation inconsistent with other lines in this file, I think your lines use tabs and the rest use spaces